### DISCUSSION TITLE
Opening same workspace on another device (including terminals etc)

### DISCUSSION THREAD

**Discussion by drmrbrewer** [coder/code-server#5812] [Q&A]
Occasionally, when away from my normal workstation, I'd like to be able to hop on to my code-server workspace to do a few bits and pieces.

But I find that I don't see exactly the same workspace setup as the one when I'm at my desk, specifically I don't have the existing set of terminals already open and usable.

Are these terminal settings stored locally in the browser, e.g. using cookies or [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) or something along those lines, i.e. only accessible to a specific browser?

**Comment by jsjoeio** [comment]
I believe so! At least from the last time I tested, they were. I don't know what a good workaround would be unfortunately 🤔

**Reply by drmrbrewer** [reply]
OK, that's a shame but I guess it is what it is!

**Reply by jsjoeio** [reply]
I know :/ Could be worth raising upstream because I'd like it to work as you explained as well!

**Reply by drmrbrewer** [reply]
@jsjoeio I [asked the question](https://github.com/microsoft/vscode/issues/167831) over at vscode, and indeed the "state is stored on the client side" by design.  There is "settings sync feature" which allows you to "roam state", but it doesn't cover terminals.  So my ideal of being able to open exactly the same workspace (including having the same terminals and associated sessions open) just isn't possible.

**Reply by jsjoeio** [reply]
Darn! Well, I'm glad you asked at least. Maybe they'll include it one day. I agree, it would be nice!

**Comment by tobycm** [comment]
I stumbled across "Terminal: Attach to Session" command today and I think you should try it out

**Reply by lg-nineoneninethreenine** [reply]
Attach to session? Doesn't that mean it adds to the client than the server?

**Reply by tobycm** [reply]
I think this command attach a terminal session on the server to your client

**Reply by lg-nineoneninethreenine** [reply]
Well, to one client, right? I don't think it would be that useful. If you are able to share localStorage around though, I think you would be in luck then.

**Reply by jsjoeio** [reply]
Interesting! Let us know how it works out!

**Reply by suzakuwcx** [reply]
Great! This solve my problem. This command can clean up those opening terminals which created in different device, just attach in by this command and exit manual, so that the "ps" command will not leave tons of "/bin/bash" process on it
